### PR TITLE
Add webhook to refund payments

### PR DESCRIPTION
**Pull request by Naokimi**
Fixes https://github.com/nebulab/solidus_bolt/issues/41

Full refund:
image

Partial refund:
image

### CODE REVIEW

**Review by DanielePalombo — CHANGES_REQUESTED** [review]

File: app/webhooks/solidus_bolt/handlers/credit_handler.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module SolidusBolt
+  module Handlers
+    class CreditHandler < BaseHandler
+      def call
+        response_code = params[:data][:reference]
```
Comment (DanielePalombo) [inline app/webhooks/solidus_bolt/handlers/credit_handler.rb:7]: Reading the payload I think it should be

`response_code = params[:data][:source_transaction][:reference]`

File: app/services/solidus_bolt/payments/credit_sync_service.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        super
+      end
+
+      def call
+        Spree::Refund.create(
+          payment: payment,
+          amount: amount.to_f / 100,
+          transaction_id: response_code,
+          refund_reason_id: refund_reason
+        )
```
Comment (piyushswain) [inline app/services/solidus_bolt/payments/credit_sync_service.rb]: Shouldn't we check if a `Spree::Refund` with the `transaction_id` already exists or not?

If a `Spree::Refund` with the given `transaction_id` exists then we should let the process fail silently.

**Review by piyushswain — CHANGES_REQUESTED** [review]

File: app/services/solidus_bolt/payments/credit_sync_service.rb (hunk 2 above)
Comment (Naokimi) [reply app/services/solidus_bolt/payments/credit_sync_service.rb]: can't we issue multiple partial refunds to the same payment?

File: app/services/solidus_bolt/payments/credit_sync_service.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+      def create_refund
+        payload = {
+          payment_id: payment.id,
+          amount: amount.to_f / 100,
+          transaction_id: transaction_id,
+          refund_reason_id: refund_reason
+        }
+
+        if Gem.loaded_specs['solidus'].version >= Gem::Version.create('3.0')
```
Comment (DanielePalombo) [inline app/services/solidus_bolt/payments/credit_sync_service.rb:34]: 🤔 
Why you didn't use `Spree.solidus_gem_version`?

**Review by DanielePalombo — APPROVED** [review]
LGTM. I have left only one minor feedback

File: app/services/solidus_bolt/payments/credit_sync_service.rb (hunk 3 above)
Comment (Naokimi) [reply app/services/solidus_bolt/payments/credit_sync_service.rb]: fixed in 2e85755

File: app/services/solidus_bolt/payments/credit_sync_service.rb (hunk 2 above)
Comment (Naokimi) [reply app/services/solidus_bolt/payments/credit_sync_service.rb]: fixed 4317db294a8588c40f43ff0e66d4f8e4285465f2

**Pull request merged**